bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -60,7 +60,7 @@
             '0', '1', '2', '3', '4', '5', '6', '7',
             '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
 
-    private final int timestamp;
+    private final long timestamp;
```

Comment:
As discussed in person, see if we can keep this a long and do something like with the UnsignedLongs class to treat it as unsigned.

Reply:
I have added an `UnsignedInts` class that emulates the `UnsignedLongs` class.